projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -0,0 +1,330 @@
+import {
+    AfterContentInit,
+    AfterViewInit,
+    Component,
+    ContentChildren,
+    EventEmitter,
+    HostBinding,
+    Input,
+    OnDestroy,
+    Output,
+    QueryList
+} from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { ACCORDION_NAVIGATION_KEYS } from '../core/utils';
+import { IExpansionPanelCancelableEventArgs, IExpansionPanelEventArgs } from '../expansion-panel/expansion-panel.common';
+import { IgxExpansionPanelComponent } from '../expansion-panel/expansion-panel.component';
+import { ToggleAnimationSettings } from '../expansion-panel/toggle-animation-component';
+
+let NEXT_ID = 0;
+
+@Component({
+    selector: 'igx-accordion',
+    templateUrl: 'accordion.component.html'
+})
+export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, OnDestroy {
+    /**
+     * Sets/gets the `id` of the accordion component.
+     * If not set, `id` will have value `"igx-accordion-0"`;
+     * ```html
+     * <igx-accordion id="my-first-accordion"></igx-expansion-panel>
+     * ```
+     * ```typescript
+     * const accordionId = this.accordion.id;
+     * ```
+     *
+     * @memberof IgxAccordionComponent
+     */
+    @HostBinding('attr.id')
+    @Input()
+    public id = `igx-accordion-${NEXT_ID++}`;
+
+    /**
+     * The role attribute of the accordion.
+     **/
+    @HostBinding('attr.role')
+    public role = 'accordion';
```

Comment:
Just noticed this in a previous comment - fairly certain there's no such role defined in the WAI-ARIA spec, so this should be removed. Looking at the [practices](https://www.w3.org/TR/wai-aria-practices/#accordion) info the requirements are for headers to be buttons linked to the content and as far as I can tell the expansion panel already covers that? The only specific thing I saw was a suggestion to wrap the panels in a list (outside of suggested practices) to aid in navigation/orientation, though I think using headings on our end should serve a similar purpose.